src/main/java/seedu/rc4hdb/storage/csv/CsvReader.java

```diff
@@ -0,0 +1,141 @@
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.rc4hdb.commons.core.LogsCenter;
+import seedu.rc4hdb.commons.exceptions.DataConversionException;
+import seedu.rc4hdb.commons.util.FileUtil;
+import seedu.rc4hdb.logic.parser.ParserUtil;
+import seedu.rc4hdb.logic.parser.exceptions.ParseException;
+import seedu.rc4hdb.model.ReadOnlyResidentBook;
+import seedu.rc4hdb.model.ResidentBook;
+import seedu.rc4hdb.model.resident.Resident;
+import seedu.rc4hdb.model.resident.fields.Email;
+import seedu.rc4hdb.model.resident.fields.Gender;
+import seedu.rc4hdb.model.resident.fields.House;
+import seedu.rc4hdb.model.resident.fields.MatricNumber;
+import seedu.rc4hdb.model.resident.fields.Name;
+import seedu.rc4hdb.model.resident.fields.Phone;
+import seedu.rc4hdb.model.resident.fields.Room;
+import seedu.rc4hdb.model.resident.fields.Tag;
+import seedu.rc4hdb.storage.csv.exceptions.InvalidCsvFileFormatException;
+
+/**
+ * Reads CSV files into {@code ResidentBook} format.
+ */
+public class CsvReader {
+
+    private static final Logger logger = LogsCenter.getLogger(CsvReader.class);
+
+    private static final Pattern CSV_FORMAT = Pattern.compile("(?<Name>[^,]+),(?<Phone>[^,]+),(?<Email>[^,]+),"
+            + "(?<Room>[^,]+),(?<Gender>[^,]+),(?<House>[^,]+),(?<Matric>[^,]+),(?<Tags>[^,]+)");
+
+    /**
+     * Reads resident data from CSV file into ResidentBook format.
+     * @param filePath the file path of the CSV file to be read.
+     * @return a resident book created from resident data from the CSV file to be read.
+     * @throws IOException if something unexpected occurs while reading the file.
+     * @throws DataConversionException if the CSV file is in an invalid format.
+     */
+    public Optional<ReadOnlyResidentBook> readCsvFile(Path filePath) throws IOException, DataConversionException {
+        requireNonNull(filePath);
+        assert(isCsvFilePath(filePath));
+        if (!FileUtil.isFileExists(filePath)) {
+            logger.warning("File does not exist: " + filePath);
+            return Optional.empty();
+        }
+
+        String csvAsString = cleanBom(filePath);
+        List<String> lines = splitLines(csvAsString);
+        ResidentBook residentBook = new ResidentBook();
+
+        for (String line : lines) {
+            residentBook.addResident(readLine(line));
+        }
+
+        return Optional.of(residentBook);
+    }
+
+    private boolean isCsvFilePath(Path filePath) {
+        return filePath.toString().endsWith(".csv");
+    }
+
+    /**
+     * Splits the CSV string by lines and return as a list of lines.
+     */
+    private List<String> splitLines(String csvAsString) {
+        Scanner sc = new Scanner(csvAsString);
+        List<String> lines = new ArrayList<>();
+        while (sc.hasNext()) {
+            lines.add(sc.nextLine());
+        }
+        return lines;
+    }
+
+    /**
+     * Cleans up the BOM that exists in some UTF-8 files.
```

Comment:
Just a nitpick but you probably would want to explain what the clean-up is exactly. I feel that just saying clean up is a bit vague.